security-baseline/rule-types/github/osps-le-04.yaml

```
@@ -0,0 +1,90 @@
+version: v1
+release_phase: alpha
+type: rule-type
+name: osps-le-04
+display_name: Released software assets have OSI/FSF approved license
+short_failure_message: Released software assets do not have an OSI/FSF approved license
+severity:
+  value: info
+context:
+  provider: github
+description: |
+  Verifies that released software assets include a license file that meets
+  the OSI Open Source Definition or the FSF Free Software Definition.
+guidance: |
+  Ensure that your releases include a LICENSE or COPYING file that contains
+  an OSI-approved or FSF-approved license. Common examples include:
+  - Apache License 2.0
+  - BSD License
+  - GNU General Public License
+  - MIT License
+  
+  For a complete list of approved licenses, see:
+  - OSI: https://opensource.org/licenses/
+  - FSF: https://www.gnu.org/licenses/license-list.html
+def:
+  in_entity: release
+  rule_schema: { }
+  ingest:
+    type: rest
+    rest:
+      endpoint: "/repos/{{ mapGet .Entity.Properties \"github/owner\" }}/{{ mapGet .Entity.Properties \"github/repo\" }}/releases/{{ mapGet .Entity.Properties \"upstream_id\" }}/assets"
+      parse: json
+  eval:
+    type: rego
+    data_sources:
+      - name: spdx
+    rego:
+      type: constraints
+      def: |
+        package minder
+        
+        import rego.v1
+        
+        license_pattern := "^license(\\..*)?$"
+        is_license_file(name) if {
+          regex.match(license_pattern, lower(name))
+        }
+
+        violations[{"msg": msg}] {
+          license_file := asset {
+            some asset in input.ingested
+            is_license_file(asset.name)
+          }
```

This seems simpler as:

```suggestion
        violations[{"msg": msg}] {
          license_file := asset {
            some asset in input.ingested
            startswith(lower(name), "license")
          }
```